libraries/auth/src/main/java/org/datatransferproject/auth/OAuth2DataGenerator.java

```diff
@@ -116,13 +113,26 @@ private AuthorizationCodeFlow createFlow() {
         BearerToken.authorizationHeaderAccessMethod(), // Access Method
         httpTransport,
         new JacksonFactory(),
-        new GenericUrl(tokenServerUrl),
+        new GenericUrl(config.getTokenUrl()),
         new ClientParametersAuthentication(
             clientId, clientSecret), // HttpExecuteInterceptor
         clientId, // client ID
-        authServerUrl)
+        config.getAuthUrl())
         .setScopes(scopes);
 
     return authCodeFlowBuilder.build();
   }
+
+  private void validateConfig() {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.getServiceName()),
+        "Config is missing service name");
+    Preconditions
+        .checkArgument(!Strings.isNullOrEmpty(config.getAuthUrl()), "Config is missing auth url");
+    Preconditions
+        .checkArgument(!Strings.isNullOrEmpty(config.getTokenUrl()), "Config is missing token url");
+    Preconditions
```

Comment:
same here about scopes for OAuth2, but please double check in case I'm wrong

Reply:
Looks like scopes are not required, as evidence by blog posts saying they shouldn't always be used.

Reply:
same here re documenting scope decision